### PR TITLE
Closes #2442 Fixed advs template and vignette

### LINKED ISSUE #2442: Reconsider presence/need of AVALC in ADaMs with only numeric parameters
### Background Information

Originally discussed [here](https://github.com/pharmaverse/admiral/pull/2402#issuecomment-2069865760).

According to CDISC guidance (3.3.4.2 of the ADaMIG v1.3), ADaMs with only numeric parameters should not have AVALC as a variable. Currently {admiral} does not follow this guidance in ADVS (and maybe other ADaMs too).

@nbrucken17 tagging you for awareness - thanks for flagging.
@Fanny-Gautier @Lina2689 tagging you in case this is relevant for {admiralpeds}.
@kaz462 tagging you as this will require an update to the {pharmaverseadam} labelling spec if any vars are removed.

### Definition of Done

- [ ] Review ADVS template and vignette and identify whether we are actually following CDISC guidance re: AVALC. If not needed, remove.
- [ ] Identify impact of removing AVALC - are we breaking any checks/codelists?
- [ ] Identify any other ADaMs where same discussion applies - e.g. ADLB?

### PR DESCRIPTION
**Pull request by StefanThoma**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/bds_finding.Rmd
Comment (StefanThoma) [inline vignettes/bds_finding.Rmd:1]: Not sure if the reference to AVALC in the header on line 259 and the following text should also be removed.

**Comment by StefanThoma** [comment]
In the ad_adpp template: 

`pp$PPSTRESC` is not a character variable but of type `double`. 
all.equal(pp$PPSTRESC, pp$PPSTRESN) is TRUE except for attributes.
Is this an error?

All values are numeric, so ADPP AVALC is also all numeric.
Therefore, AVALC should not be mapped in the template, right?
Or would we expect non-numeric information in pp datasets in general?

**Comment by manciniedoardo** [comment]
> @bms63 & @manciniedoardo In the ad_adpp template:
> 
> `pp$PPSTRESC` is not a character variable but of type `double`. all.equal(pp$PPSTRESC, pp$PPSTRESN) is TRUE except for attributes. Is this an error?
> 
> All values are numeric, so ADPP AVALC is also all numeric. Therefore, AVALC should not be mapped in the template, right? Or would we expect non-numeric information in pp datasets in general?

@jeffreyad any thoughts?

**Comment by bms63** [comment]
> @bms63 & @manciniedoardo In the ad_adpp template:
> 
> `pp$PPSTRESC` is not a character variable but of type `double`. all.equal(pp$PPSTRESC, pp$PPSTRESN) is TRUE except for attributes. Is this an error?
> 
> All values are numeric, so ADPP AVALC is also all numeric. Therefore, AVALC should not be mapped in the template, right? Or would we expect non-numeric information in pp datasets in general?

AVALC should always be character.  Perhaps this is a mistake in the PP dataset.

**Comment by jeffreyad** [comment]
> > @bms63 & @manciniedoardo In the ad_adpp template:
> > `pp$PPSTRESC` is not a character variable but of type `double`. all.equal(pp$PPSTRESC, pp$PPSTRESN) is TRUE except for attributes. Is this an error?
> > All values are numeric, so ADPP AVALC is also all numeric. Therefore, AVALC should not be mapped in the template, right? Or would we expect non-numeric information in pp datasets in general?
> 
> @jeffreyad any thoughts?

Thanks, I put in a PR to update `pp$PPSTRESC`

https://github.com/pharmaverse/pharmaversesdtm/pull/103

**Comment by bms63** [comment]
Thanks @jeffreyad!! we might need to use remotes for pharmaversesdtm @StefanThoma to get the latest once it is merged in.

**Comment by StefanThoma** [comment]
> Thanks @jeffreyad!! we might need to use remotes for pharmaversesdtm @StefanThoma to get the latest once it is merged in.

For ADPP I just remove AVALC altogether.

File: inst/templates/ad_adeg.R [hunk 1]
```diff
@@ -251,7 +254,7 @@ adeg <- adeg %>%
       PARAMCD != "EGINTP"
     )
   )
-
+adlb$LBBLFL
```
Comment (StefanThoma) [inline inst/templates/ad_adeg.R:257]:
```suggestion
```

**Comment by StefanThoma** [comment]
> Something amiss with the code style and the templates

Thanks, should be good now!

File: NEWS.md [hunk 2]
```diff
@@ -59,7 +65,9 @@ functions and some of the more advanced topics have been moved to the new "Progr
  
 ## Various
 
- - The function `dplyr::transmute()` is superseded in favor of `dplyr::mutate(.keep = "none")`. Consequently, all the admiral functions that utilized the former have been updated accordingly. (#2274)
+- Templates and vignettes do not add or populate `AVALC` for findings BDS where the information contained in `AVALC` would be redundant with `AVAL`. (#2442)
```
Comment (bms63) [inline NEWS.md:68]:
```suggestion
- Templates and vignettes do not add or populate `AVALC` for BDS-findings datasets where the information contained in `AVALC` would be redundant with `AVAL`. (#2442)
```

**Comment by bms63** [comment]
Thanks again @StefanThoma for jumping in on this one!!

**Pull request merged**